internal/_posts/2015-12-03-dont-go-chasing-waterfalls.markdown

```
@@ -0,0 +1,93 @@
+
+### "Chinese Whispers"
+As mentioned already, however good the analyst is at cascading knowledge down to the developers, something is always lost in the translation. The further a role is from the actual requirements, the less that role understands those requirements. So, by the time the product owner has told the analyst a requirement, the analyst has conveyed it to the developers, and the developers have handed it over to the testers, the testers will probably have little idea what the original requirement was, and how to test it. Typically they will end up testing what has been built, not what was required.
+
+### The team is not equally empowered
+Typically, in these situations the testers have a kind of "sign-off"; they find bugs and stop software being released because of them, until the bugs have been triaged and seen to be acceptable for release or fixed. This gives them unreasonable power, when, in fact, the whole team should be deciding on the quality of what is released.
+
+### The team is not equally responsible
+When there are dedicated, non-developer roles in a team, the team is no longer collectively responsible for the quality of the product. Developers develop, then hand responsibility to the testers. This kind of "over-the-wall" thinking means that the team as a whole is no longer collectively responsible - developers won't look at bugs that they deem un-reproducible, and testers will complain that developers haven't done enough basic testing before handing over the code. 
+
+I have seen this in many teams.
+
+### Waterfall stops failing early
+A guiding principle of Agile is "fail early"; in fact, as early as possible. Don't wait until code has been written and handed over to a test team before bugs are discovered. Bugs should be identified as soon as code is checked in (*or earlier, if possible?*). Otherwise, you may be adding code that relies on code that already contains bugs. Very dangerous.
```

"fail fast" is probably a more commonly used term, but up to you :)